src/error.rs

```diff
@@ -69,5 +69,4 @@ pub enum RoutingError {
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub enum BootstrapResponseError {
```

Comment:
probably no need to make this as `enum` anymore ? 
as you will only be `NotApproved`